snapcraft/parts/yaml_utils.py

```diff
@@ -246,6 +246,12 @@ def get_snap_project(project_dir: Path | None = None) -> _SnapProject:
             (project_dir / snap_project.project_file).resolve(strict=True)
         except FileNotFoundError:
             continue
+        except NotADirectoryError:
```

Comment:
Thanks for this! Could you add a test for this error case along the lines of the [success case test](https://github.com/canonical/snapcraft/blob/972842590d2cc787282e76c9b08ba8fa962b7a31/tests/unit/parts/test_yaml_utils.py#L248-L257)?